### ISSUE TITLE
Fleky Test: Scheduler Suite: [It] Preemption When most quota is in a shared ClusterQueue in a cohort should allow preempting workloads while borrowing

### ISSUE THREAD

**Issue by tenzen-y** [kubernetes-sigs/kueue#2100]
/kind flake

**What happened**:
Failure on `Scheduler Suite: [It] Preemption When most quota is in a shared ClusterQueue in a cohort should allow preempting workloads while borrowing`.

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/2081/pull-kueue-test-integration-release-0-6/1783900768210784256

```shell
{Expected success, but got an error:
    <*errors.StatusError | 0xc000ef1040>: 
    resourceflavors.kueue.x-k8s.io "one" already exists
    {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {
                SelfLink: "",
                ResourceVersion: "",
                Continue: "",
                RemainingItemCount: nil,
            },
            Status: "Failure",
            Message: "resourceflavors.kueue.x-k8s.io \"one\" already exists",
            Reason: "AlreadyExists",
            Details: {
                Name: "one",
                Group: "kueue.x-k8s.io",
                Kind: "resourceflavors",
                UID: "",
                Causes: nil,
                RetryAfterSeconds: 0,
            },
            Code: 409,
        },
    } failed [FAILED] Expected success, but got an error:
    <*errors.StatusError | 0xc000ef1040>: 
    resourceflavors.kueue.x-k8s.io "one" already exists
    {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {
                SelfLink: "",
                ResourceVersion: "",
                Continue: "",
                RemainingItemCount: nil,
            },
            Status: "Failure",
            Message: "resourceflavors.kueue.x-k8s.io \"one\" already exists",
            Reason: "AlreadyExists",
            Details: {
                Name: "one",
                Group: "kueue.x-k8s.io",
                Kind: "resourceflavors",
                UID: "",
                Causes: nil,
                RetryAfterSeconds: 0,
            },
            Code: 409,
        },
    }
In [BeforeEach] at: /home/prow/go/src/sigs.k8s.io/kueue/test/integration/scheduler/preemption_test.go:496 @ 04/26/24 16:55:22.651

There were additional failures detected after the initial failure. These are visible in the timeline
}
```

**What you expected to happen**:
No errors happen.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by IrvingMg** [comment]
@tenzen-y This issue seems to be caused because we're missing the deletion of the resource flavor after each test. Here we're creating it 
https://github.com/kubernetes-sigs/kueue/blob/8127d46d54febb2d5fd5f5a19ed32e0fc8206099/test/integration/scheduler/preemption_test.go#L417 

but then, in `AfterEach` that object isn't deleted.

https://github.com/kubernetes-sigs/kueue/blob/8127d46d54febb2d5fd5f5a19ed32e0fc8206099/test/integration/scheduler/preemption_test.go#L446 

However, this was fixed already in #1930. Should we cherry pick that commit?

**Comment by tenzen-y** [comment]
> @tenzen-y This issue seems to be caused because we're missing the deletion of the resource flavor after each test. Here we're creating it
> 
> https://github.com/kubernetes-sigs/kueue/blob/8127d46d54febb2d5fd5f5a19ed32e0fc8206099/test/integration/scheduler/preemption_test.go#L417
> 
> but then, in `AfterEach` that object isn't deleted.
> 
> https://github.com/kubernetes-sigs/kueue/blob/8127d46d54febb2d5fd5f5a19ed32e0fc8206099/test/integration/scheduler/preemption_test.go#L446
> 
> However, this was fixed already in #1930. Should we cherry pick that commit?

@IrvingMg Oh, good catch! Thank you for investigating this!
Let me cherry-pick that PR by prow.

**Comment by tenzen-y** [comment]
This would be resolved by #2103 
/close
